pkg/scte35/stream.go

```diff
@@ -0,0 +1,354 @@
+// Copyright 2022 Adrian of Doom
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or   implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+package scte35
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+const (
+    // pktSz is the size of an MPEG-TS packet in bytes.
+    pktSz = 188
+
+    // bufSz is the size of a read when parsing files.
+    bufSz = 13000 * pktSz
+)
+
+type PacketData struct {
+	PacketNumber int     `json:",omitempty"`
+	Pid          uint16  `json:",omitempty"`
```

Comment:
Related to above comment, `PID`, `PCR`, `PTS` etc

Reply:
I can rename functions, but I can't do that with local vars and such.

Reply:
```lua
// PIDs holds collections of PIDs by type for scte35.Stream.
type PIDs struct {
	PMTPIDs    []uint16
	PCRPIDs   []uint16
	Scte35PIDs []uint16
}

func (p *PIDs) isPMTPID(pid uint16) bool {
	return isIn16(p.PMTPIDs, pid)
}

func (p *PIDs) addPMTPID(pid uint16) {
	if !p.isPMTPID(pid) {
		p.PMTPIDs = append(p.PMTPIDs, pid)
	}
}
```

Does that  looks kind  of kooky?